lib/graphql/schema/printer.rb

```diff
@@ -44,6 +48,10 @@ def print_schema_definition(schema)
       BUILTIN_SCALARS = Set.new(["String", "Boolean", "Int", "Float", "ID"])
       private_constant :BUILTIN_SCALARS
 
+      def is_spec_directive(directive)
+        ['skip', 'include', 'deprecated'].include?(directive.name)
+      end
```

Comment:
> would be surprised not to see `@defer` / `@stream` in the query dump 

Just to make sure we're on the same page. You'd still see `@defer` and `@stream` on the fields, you just don't see their definitions, i.e. `directive @defer`.

I think the other reason they do this in the JS implementation is because `directives` can be specified on the `schema`: https://github.com/graphql/graphql-js/blob/a725499b155285c2e33647a93393c82689b20b0f/src/type/schema.js#L50-L54

We don't support that yet.